m2cgen/assemblers/linear.py

```diff
@@ -70,6 +75,111 @@ def _get_coef(self):
         return self.model.params[:self.model.k_exog]
 
 
+class StatsmodelsGLMModelAssembler(StatsmodelsLinearModelAssembler):
+
+    def _final_transform(self, ast_to_transform):
+        link_function = type(self.model.model.family.link).__name__
+        link_function_lower = link_function.lower()
+        supported_functions = {
+            "logit": self._logit,
+            "power": self._power,
+            "inverse_power": self._inverse_power,
+            "sqrt": self._sqrt,
+            "inverse_squared": self._inverse_squared,
+            "identity": self._identity,
+            "log": self._log,
+            "cloglog": self._cloglog,
+            "negativebinomial": self._negativebinomial,
+            "nbinom": self._negativebinomial
+        }
+        if link_function_lower not in supported_functions:
+            raise ValueError(
+                "Unsupported link function '{}'".format(link_function))
+        link_fun = supported_functions[link_function_lower]
+        return link_fun(ast_to_transform)
+
+    def _logit(self, ast_to_transform):
+        return utils.div(
+            ast.NumVal(1.0),
+            utils.add(
+                ast.NumVal(1.0),
+                ast.ExpExpr(
+                    utils.sub(
+                        ast.NumVal(0.0),
+                        ast_to_transform))))
+
+    def _power(self, ast_to_transform):
+        power = self.model.model.family.link.power
+        if power == 1:
+            return self._identity(ast_to_transform)
+        elif power == -1:
+            return self._inverse_power(ast_to_transform)
+        elif power == 2:
+            return ast.SqrtExpr(ast_to_transform)
```

Comment:
I'm a bit confused here. `SqrtExpr` represents a square root. Why do we use it with power 2 which I assume should mean just a squared expression.

Reply:
It appears that all powers are "inversed" and it blows my mind :D Can we perhaps inverse it at the very beginning, eg:
```
power = 1 / self.model.model.family.link.power
```
So that the rest of the code makes more sense?
Or rename the `power` variable to something more specific, eg. `inversed_power`.

Reply:
Sorry, I messed up with function names! Should be more clear now after https://github.com/BayesWitnesses/m2cgen/pull/195/commits/0fbaefb9be2a2af6ae5b39d9fbb0a955d8f973bc.
For the reference: https://github.com/statsmodels/statsmodels/blob/master/statsmodels/genmod/families/links.py.